core/lib/spree/core/state_machines/inventory_unit.rb

```diff
@@ -34,6 +34,14 @@ module InventoryUnit
             event :cancel do
               transition to: :canceled, from: ::Spree::InventoryUnit::CANCELABLE_STATES.map(&:to_sym)
             end
+
+            event :on_hand do
```

Comment:
I'm not 100% sure about this event name. 

Semantically speaking, event names should be verbs, as they actively change the state of an object. In this case, the action name is the same as the target state, and this may lead to some confusion too.  
Also, there's the query method #can_on_hand? added automatically, which again does not make 100% sense to me. 

What about changing the name to something like `set_on_hand` or `make_on_hand`? And if you find a better naming, just go for it.

Reply:
I agree with you, I was not convinced too by the name of that event but couldn't think of something better. 

I like `set_on_hand` proposed by you ❤️.

Sadly, as pointed out in the last comment of Kennyadsl, I'm probably going to close this PR to move the whole resume functionality out of Solidus.

Thanks a lot for your feedback, I'll note it down to use it when the work on the extracted functionality will start 🙏.